cparser/processor.js

```diff
@@ -41,8 +41,16 @@ module.exports = that = {
 				return false;
 			}
 
-			if ((fileBuffer.indexOf('#pragma SPARK_NO_PREPROCESSOR') >= 0) ||
-				(fileBuffer.indexOf('#pragma PARTICLE_NO_PREPROCESSOR') >= 0) ||
+			var noPreprocessorMatch = fileBuffer.match('#pragma (SPARK_NO_PREPROCESSOR|PARTICLE_NO_PREPROCESSOR)');
+			if (noPreprocessorMatch) {
```

Comment:
I'm sure that `#pragma PARTICLE_NO_PREPROCESSOR` avoids including application.h. This simple code does not compile:

```
#pragma PARTICLE_NO_PREPROCESSOR
int foo = 42;
void setup() {
  Particle.function("foo", foo);
}
```